lib/util/esWrapper.js

```diff
@@ -25,74 +25,89 @@ const
     regex: /^\[illegal_argument_exception] \[(.*?)] is defined as an object in mapping \[(.*?)] but this name is already used for a field in other types$/,
-    replacement: 'Can not set mapping for field "$1" on collection "$2" because the field name is already used in another collection with a different type'
+    subcode: 'duplicate_field_name',
+    getPlaceholders: matches => [matches[1], matches[2]]
   },
   {
     // [illegal_argument_exception] mapper [source.flags] of different type, current_type [string], merged_type [long]
     regex: /^\[illegal_argument_exception] mapper \[(.*?)] of different type, current_type \[(.*?)], merged_type \[(.*?)]$/,
-    replacement: 'Can not change type of field "$1" from "$2" to "$3"'
+    subcode: 'invalid_type_change',
+    getPlaceholders: matches => [matches[1], matches[2], matches[3]]
   },
   {
     // [mapper_parsing_exception] Mapping definition for [flags] has unsupported parameters:  [index : not_analyzed]
     // eslint-disable-next-line no-regex-spaces
     regex: /^\[mapper_parsing_exception] Mapping definition for \[(.*?)] has unsupported parameters: \[(.*?)]$/,
-    replacement: 'Parameter "$2" is not supported for field "$1"'
+    subcode: 'unsupported_parameter_for_field',
+    getPlaceholders: matches => [matches[2], matches[1]]
   },
   {
     // [mapper_parsing_exception] No handler for type [booleasn] declared on field [not]
     regex: /^\[mapper_parsing_exception] No handler for type \[(.*?)] declared on field \[(.*?)]$/,
-    replacement: 'Can not set mapping for field "$2" because type "$1" does not exist'
+    subcode: 'type_does_not_exist',
+    getPlaceholders: matches => [matches[2], matches[1]]
   },
   {
     // [mapper_parsing_exception] failed to parse [conditions.host.flags]
     regex: /^\[mapper_parsing_exception] failed to parse \[(.*?)]$/,
-    replacement: 'Failed to validate value of field "$1". Are you trying to insert nested value in a non-nested field ?'
+    subcode: 'fail_to_parse_field',
+    getPlaceholders: matches => [matches[1]]
+  },
+  {
+    // Failed to parse mapping [_doc]: Expected map for property [fields] on field [enabled] but got a class java.lang.String
+    regex: /^Failed to parse mapping \[.*\]: Expected \w+ for property \[.*\] on field \[(.*)\]/,
+    subcode: 'cannot_parse_mapping',
+    getPlaceholders: matches => [matches[1], matches[0]]
   },
   {
     // [index_not_found_exception] no such index, with { resource.type=index_or_alias & resource.id=foso & index=foso }
-    regex: /^\[index_not_found_exception] no such index, with { resource\.type=([^\s]+) (& )?resource\.id=([^\s]+) (& )?(index_uuid=.* )?index=([^\s]+) }$/,
-    replacement: 'Index "$3" does not exist, please create it first'
+    regex: /^no such index \[([%#])(.*)\/(.*)\]$/,
+    subcode: 'index_or_collection_not_found',
+    getPlaceholders: matches => {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove curly braces and "return" from this arrow function body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3524)